### ISSUE TITLE
Bug report: [500] Internal Server Error, can't send file to PC

### ISSUE THREAD

**Issue by daidaiworm** [localsend/localsend#461]
### Describe the bug

almost the same issue, both version Localsend 1.7.0 and 1.9.1.
1 PC: win10
2 Android Smart Phones(Mi 10 Ultra & Mi Max 3)

Both Android Phones can send/receive file to/from each other smoothly.
PC can send file to 2 phones.
Android Phones can't send file to PC.

### To reproduce

2. Click on '...'
3. Scroll down to '...'

### Expected behavior

File should be sent as expected.

### Screenshots

![微信图片_20230510213449](https://github.com/localsend/localsend/assets/3594904/840b2373-1b51-4168-aadf-1f4a31536070)

### Smartphone (please complete the following information)

- Device: Mi 10 Ultra
- OS: MIUI 12.5.6 / Android 11
- Version: 1.7.0 / 1.9.1

**Comment by Tienisto** [comment]
Do you see any reaction from the receiving device?

**Comment by daidaiworm** [comment]
> Do you see any reaction from the receiving device?

No. Nothing happens on my PC.

**Comment by daidaiworm** [comment]
I'm using 3rd party's anti-virus/firewall.
When encounting this weird issue, I turned off and quit those firewall, but failed all the same.

<img width="442" alt="zf_2023-05-12_181528" src="https://github.com/localsend/localsend/assets/3594904/9af08763-43d9-418f-88ef-19c4d8a14f0e">
<img width="225" alt="zf_2023-05-12_182020" src="https://github.com/localsend/localsend/assets/3594904/faf207dc-bc13-47b1-b39a-716d06cbfb96">

**Comment by Tienisto** [comment]
Yes, error 500 indicates that the receiving device (Windows) did received the request but somehow it failed to handle it.

There might be an unusual setup which hasn't been handled yet.

Do you have a Downloads folder? Can you select another destination path in the settings?

**Comment by daidaiworm** [comment]
I prefer portable version, so I downloaded 1.9.1_window.
When failed to receive, I tried different version on my pc
1.9.1_msix
1.9.0_window
1.9.0_msix

As for the path, I just changed from "D:\soft\LocalSend" to "D:\temp\LocalSend"
Issue remain same.
<img width="214" alt="zf_2023-05-12_234025" src="https://github.com/localsend/localsend/assets/3594904/558086a2-ae29-4009-9ffc-aab0bb7d4291">

**Comment by daidaiworm** [comment]
Tried the latest version 1.10.0, both portable and msix version failed receive.

I tested with another 2 pc. Let's call my issued pc as mainPC.
Result:
2 android phones, 2 win10 pc can send file to each other.
None phone and pc can send file to mainPC.

I've tried every version listed on github, from 1.7.0 to 1.10.0, both portable and msix version.

Since there is notice on the homepage
**Due to certificate changes, you need to uninstall v1.7.0 or earlier before installing v1.8.0 or later.**
Is there anyway to uninstall that certificate manully?

**Comment by guemax** [comment]
Just wanted to add more context to this issue, as I experienced it as well: 

> @Tienisto: Yes, error 500 indicates that the receiving device (Windows) did received the request but somehow it failed to handle it.

Running Localsend from a console prints the following error message:
```
flutter: Server started. (Port: 53317, HTTPS only)
flutter: Bind UDP multicast port (ip: [192.168.2.20], group: 224.0.0.167, port: 53317)
flutter: [ANNOUNCE/UDP]
flutter: [ANNOUNCE/UDP]
flutter: [ANNOUNCE/UDP]
flutter: [DISCOVER/UDP] Fantastic Blueberry (192.168.2.23, model: Samsung)
flutter: [RESPONSE/TCP] Announcement of Fantastic Blueberry (192.168.2.23, model: Samsung) via TCP
ERROR - 2023-08-29 20:30:10.152892
POST /api/localsend/v2/prepare-upload
Error thrown by handler.
Null check operator used on a null value
package:shelf/shelf_io.dart 138  handleRequest
```
Please note that both UDP and TCP ports 53317 are open:
```shell
$ ss -lntu
Netid State  Recv-Q Send-Q Local Address:Port    Peer Address:Port Process
udp   UNCONN 0      0            0.0.0.0:68           0.0.0.0:*    
udp   UNCONN 0      0            0.0.0.0:53317        0.0.0.0:*    
udp   UNCONN 0      0                  *:546                *:*    
tcp   LISTEN 0      128          0.0.0.0:53317        0.0.0.0:*    
```
My system information:

* Laptop: Dell Inspiron 14 7000 with NixOS 23.05 and Localsend 1.9.1
* Smartphone: Samsung S5 Mini with LineageOS 16.0 and Localsend 1.10.0

**Comment by Sqbika** [comment]
Looking into the issue, looks like the fault lies in the `lib/util/native/get_destination_directory.dart:17:50`.

The `getDownloadsDirectory()` can return null in case it cannot find the download directory,
In my case the `XDG_DOWNLOAD_DIR` env var was not defined, thus not working the way it was expecting it to.

This solved my case, but not sure if it solves other people's case. If you set the Destination to something other than the default, does the issue occur?

**Comment by Tienisto** [comment]
Regarding the fix: Maybe we can provide a fallback like `~/Downloads`? Or just `~`?

**Comment by guemax** [comment]
@Sqbika Thank you so much! Manually setting the destination did the trick for me. :tada: :smile: :tada: 

@Tienisto I am only using a minimal installation, so during the installation only the home folder is created. In my opinion, creating a fallback to `~/Downloads` is great in most cases, but it might even be necessary to fall back to `~` when this folder is not available.

**Comment by Sqbika** [comment]
Currently my solution would be a 500 response, but with the message to Sender, that the Destination is not found, so they know what to search for.

Optionally there could be a check at startup, that checks whether the Destination is actually available and notify the user that it needs attention.

**Comment by guemax** [comment]
Most of the time, `~/Downloads` will be fine. But I really like the idea of notifying the user of an unavailable destination at startup, that would've been really helpful for me... :wink:
